### ISSUE TITLE
Menu component exception

### ISSUE THREAD

**Issue by nikolay-turpitko** [matsp/material-components-vue#126]
Hi! I have a strange error during initialization of `<m-menu>` with more then one list item (with just one list item it works).

Here is my layout:

                    <m-menu-anchor slot="actions">
                        <m-icon icon="more_vert" @click="menuOpen=true"></m-icon>
                        <m-menu v-model="menuOpen">
                            <m-list>
                                <m-list-item @click="Refresh">
                                    <m-icon slot="graphic" icon="refresh"></m-icon>
                                    <span>Refresh</span>
                                </m-list-item>
                                <m-list-divider></m-list-divider>
                                <m-list-item @click="Add">
                                    <m-icon slot="graphic" icon="face"></m-icon>
                                    <span>Add User</span>
                                </m-list-item>
                            </m-list>
                        </m-menu>
                    </m-menu-anchor>

Here is an error: 

```
jsdelivr-separator.js:1 TypeError: Cannot read property 'indexOf' of undefined
    at menu.min.js:66
    at Array.filter (<anonymous>)
    at o.updateSlot (menu.min.js:66)
    at o.mounted (menu.min.js:66)
    at _t (jsdelivr-separator.js:1)
    at Object.insert (jsdelivr-separator.js:1)
    at A (jsdelivr-separator.js:1)
    at hn.__patch__ (jsdelivr-separator.js:1)
    at hn._update (jsdelivr-separator.js:1)
    at hn.<anonymous> (jsdelivr-separator.js:1)
    at St.get (jsdelivr-separator.js:1)
    at new St (jsdelivr-separator.js:1)
    at hn.$mount (jsdelivr-separator.js:1)
    at hn.$mount (jsdelivr-separator.js:1)
    at hn._init (jsdelivr-separator.js:1)
    at new hn (jsdelivr-separator.js:1)
    ...
```

Sorry, I'm using minified version from CDN, but menu component folder contains only one `indexOf` occurrence, should be there: https://github.com/matsp/material-components-vue/blob/master/components/menu/Menu.vue#L78.

**Comment by matsp** [comment]
Thanks for filling the issue @nikolay-turpitko! Best time to fix and refactor that ugly piece of code :see_no_evil:

**Comment by matsp** [comment]
I can't reproduce that error. For further investigation I would need access to an environment where I can reproduce the error.

What version you are using?

**Comment by nikolay-turpitko** [comment]
Hmm...  I include components like below. I'm trying to avoid `nodejs/npm/webpack/...` in this case. To be honest, I'm trying to avoid javascript at all :-), I'm using Go-to-JS compiler (`gopherjs`), so things are a bit rough at my setup.

```
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/combine/npm/material-components-vue@0.23.0/dist/button/button.min.css,npm/material-components-vue@0.23.0/dist/checkbox/checkbox.min.css,npm/material-components-vue@0.23.0/dist/dialog/dialog.min.css,npm/material-components-vue@0.23.0/dist/drawer/drawer.min.css,npm/material-components-vue@0.23.0/dist/elevation/elevation.min.css,npm/material-components-vue@0.23.0/dist/fab/fab.min.css,npm/material-components-vue@0.23.0/dist/floating-label/floating-label.min.css,npm/material-components-vue@0.23.0/dist/list/list.min.css,npm/material-components-vue@0.23.0/dist/line-ripple/line-ripple.min.css,npm/material-components-vue@0.23.0/dist/layout-grid/layout-grid.min.css,npm/material-components-vue@0.23.0/dist/menu/menu.min.css,npm/material-components-vue@0.23.0/dist/textfield/textfield.min.css,npm/material-components-vue@0.23.0/dist/theme/theme.min.css,npm/material-components-vue@0.23.0/dist/top-app-bar/top-app-bar.min.css,npm/material-components-vue@0.23.0/dist/typography/typography.min.css,npm/vue-good-table@2.6.2/dist/vue-good-table.min.css"/>

<script src="https://cdn.jsdelivr.net/combine/npm/vue@2.5.16/dist/vue.min.js,npm/material-components-vue@0.23.0/dist/button/button.min.js,npm/material-components-vue@0.23.0/dist/checkbox/checkbox.min.js,npm/material-components-vue@0.23.0/dist/dialog/dialog.min.js,npm/material-components-vue@0.23.0/dist/drawer/drawer.min.js,npm/material-components-vue@0.23.0/dist/elevation/elevation.min.js,npm/material-components-vue@0.23.0/dist/fab/fab.min.js,npm/material-components-vue@0.23.0/dist/floating-label/floating-label.min.js,npm/material-components-vue@0.23.0/dist/icon/icon.min.js,npm/material-components-vue@0.23.0/dist/list/list.min.js,npm/material-components-vue@0.23.0/dist/line-ripple/line-ripple.min.js,npm/material-components-vue@0.23.0/dist/layout-grid/layout-grid.min.js,npm/material-components-vue@0.23.0/dist/menu/menu.min.js,npm/material-components-vue@0.23.0/dist/textfield/textfield.min.js,npm/material-components-vue@0.23.0/dist/theme/theme.min.js,npm/material-components-vue@0.23.0/dist/top-app-bar/top-app-bar.min.js,npm/material-components-vue@0.23.0/dist/typography/typography.min.js,npm/vue-good-table@2.6.2/dist/vue-good-table.min.js"></script>
```

**Comment by nikolay-turpitko** [comment]
Ah, it may be relevant that I'm trying to anchor menu to the icon in the `<m-top-app-bar>`.

I'm on Chromium Version 66.0.3359.117 (Developer Build) built on Debian 9.4, running on Debian 9.4 (64-bit), if this important, though it usually just like Chrome.

BTW, I also noticed that if I click on menu item itself, it closes menu, but if I click on the text or icon within it, menu won't close. But, event still occur (trigger my code).

**Comment by nikolay-turpitko** [comment]
@matsp Tried Firefox 58 ESR and Opera 55 - the same behavior.

Also I noticed that only click on icon image opens menu, not on highlighted field around it, as usually icon in the top-app-bar works in other cases.

**Comment by matsp** [comment]
I will check that out. The click events are directly bound to one component. The span you use is preventing the event. I also will do some research for bubbling events.

**Comment by matsp** [comment]
@nikolay-turpitko Can you provide an example via codepen or some other sandbox environment?

**Comment by matsp** [comment]
@nikolay-turpitko Please try v0.23.7.

This issue only appears while using the library directly e.g. via CDN. I tested your code with webpack as bundler and got no errors, just on your jsfiddle..
I hope the changes fixed the problem, else please report here.

**Comment by nikolay-turpitko** [comment]
@matsp

Yeah, v0.23.7 fixes major issue - several menu items and even separator are now work!

Though there are still a couple of minor ones - like:  
- when I click on icon in the menu item, not on text, action is executed, but menu won't close;
- anchored icon (which I use to open menu) is not open menu if I click not on the text, but on the highlighted circle around it (as it should and do in other cases);

OK, for my prototype these are minor inconveniences and I'm OK with them, so I'm closing this issue, but you may want to know about them and track them separately.

Thanks!

**Comment by matsp** [comment]
@nikolay-turpitko 

thanks for feedback, I opened another issue. btw thanks for the coffee 👍

**Comment by matsp** [comment]
@nikolay-turpitko 

fixed the menu issue in #131
